include/bredis/impl/connection.ipp

```diff
@@ -111,8 +111,9 @@ void Connection<NextLayer>::write(const command_wrapper_t &command) {
 }
 
 template <typename NextLayer>
+template <typename DynamicBuffer>
 positive_parse_result_t
-Connection<NextLayer>::read(boost::asio::streambuf &rx_buff) {
+Connection<NextLayer>::read(DynamicBuffer &rx_buff) {
```

Comment:
Just a heads up, but `boost::asio:;read_until` and `boost::asio::async_read_until` only support `boost::asio::streambuf` as the type for the buffer. See:
https://github.com/boostorg/asio/blob/develop/include/boost/asio/read_until.hpp#L123

To make this really work, you have two choices:

1. Make sure you call `read_until` and `async_read_until` in a way that allows Argument Dependent Lookup to work. Users who provide their own type for **DynamicBuffer** can then supply an overload of `read_until` or `async_read_until` in the same namespace as their type, and your code will find it.

2. Provide your own implementation of `read_until` and `async_read_until` that works with any dynamic buffer. This is very easy to do, just make a copy of Asio's code and put it in your own `detail` namespace and call it instead. There's no technical reason why `read_until` and `async_read_until` can't work with a dynamic buffer concept, I think the author just hasn't gotten around to merging that change into the version of Asio used by Boost yet. This is the approach I recommend but it increases the amount of code and maintenance in your project.